examples/rust/udp-echo.rs

```diff
@@ -230,10 +231,10 @@ impl Application {
 
     #[cfg(target_os = "windows")]
     /// Converts a [sockaddr] into a [SocketAddrV4].
-    pub fn sockaddr_to_socketaddrv4(saddr: *const SOCKADDR_IN) -> Result<SocketAddrV4> {
+    pub fn sockaddr_to_socketaddrv4(saddr: *const SOCKADDR) -> Result<SocketAddrV4> {
         // TODO: Change the logic below and rename this function once we support V6 addresses as well.
 
-        let sin: SOCKADDR_IN = unsafe { *saddr };
+        let sin: SOCKADDR_IN = unsafe { *mem::transmute::<*const SOCKADDR, *const SOCKADDR_IN>(saddr) };
```

Comment:
Rather than use `mem::transmute` here (all the Rust docs I've been reading lately say to avoid `transmute` if you possibly can), couldn't you just cast `saddr` (a `*const SOCKADDR`) to a `*const SOCKADDR_IN` and then dereference that?